### DISCUSSION TITLE
Release Candidate of the Provider 91.0.2

### DISCUSSION THREAD

**Discussion by kewisch** [kewisch/gdata-provider#260] [Announcements]
Hello folks. I feel reasonably comfortable with the changes for the Provider for Google Calendar 91.0.2. This release is still based on the legacy code, but fixes a few long standing issues that have cropped up. If you've been following, I've drastically triaged the issues on the issue tracker. Here is what has changed:

### Bug Fixes

* Accept if Calendars or Tasks are disabled in G Suite ([a2bfe48](https://github.com/kewisch/gdata-provider/commit/a2bfe48ce5086f54f415db1613c7ad056880f947)), closes [#65](https://github.com/kewisch/gdata-provider/issues/65)
* Avoid asking for changes when none have been made ([8226cd7](https://github.com/kewisch/gdata-provider/commit/8226cd7297cf3508080c10bf6af2d6862e868cf3)), closes [#11](https://github.com/kewisch/gdata-provider/issues/11)
* Correctly include organizer as attendee ([f05f6c9](https://github.com/kewisch/gdata-provider/commit/f05f6c91d43d64ced5334d3b4dd3a870e136b6ee)), closes [#62](https://github.com/kewisch/gdata-provider/issues/62)
* Correctly set cookie exceptions to make logins work ([ecc6781](https://github.com/kewisch/gdata-provider/commit/ecc678123430e32ff875e672d0bbda11a1a3e5eb)), closes [#43](https://github.com/kewisch/gdata-provider/issues/43)
* Don't show the migration dialog again when cancelled ([faf2124](https://github.com/kewisch/gdata-provider/commit/faf2124e6e4e4f058e8d163efa7ce635df9d0f7a))
* Make sure sync prefs are available at startup ([18acb63](https://github.com/kewisch/gdata-provider/commit/18acb63df11d613247f080536c4a4067969f7111))
* Port bug 1687612 - Fix new account OAuth window to be E10s capable. ([44a563a](https://github.com/kewisch/gdata-provider/commit/44a563a478e47f9c31e8c686998d445824a9b0a4))
* Properly display HTML from description ([6625aba](https://github.com/kewisch/gdata-provider/commit/6625aba20513a6f579261789ae8929e086328637))
* Refresh completed/hidden tasks as well ([3a0cf1a](https://github.com/kewisch/gdata-provider/commit/3a0cf1a7ff7dc71a745766e2941b9064f678e3ce)), closes [#72](https://github.com/kewisch/gdata-provider/issues/72)
* Remove error about aRequest.name not being implemented ([3e0162e](https://github.com/kewisch/gdata-provider/commit/3e0162e108db3bc227e6c2530133d41b19538f54))
* Resolve cyclic dependency and incorrect parameters in freebusy service calls ([d5c72bb](https://github.com/kewisch/gdata-provider/commit/d5c72bbe35f57f5e06a6c0defc273abfedb9810b)), closes [#147](https://github.com/kewisch/gdata-provider/issues/147)
* Still detect code if redirect is redirected to https ([c5e1090](https://github.com/kewisch/gdata-provider/commit/c5e109068ba9c92384729aa649098657a8f211fc)), closes [#227](https://github.com/kewisch/gdata-provider/issues/227)
* Trailing slash on completion URI ([dad61c9](https://github.com/kewisch/gdata-provider/commit/dad61c9b8ba813318d3bcae749cda8b2d71c9fc2))
* Use a different port number to avoid collisions ([5a57421](https://github.com/kewisch/gdata-provider/commit/5a57421e2b13f7dec136c6722b3902aa270c6c05)), closes [#240](https://github.com/kewisch/gdata-provider/issues/240)
* Use migrated preferences instead of purged legacy prefs ([8a86dd5](https://github.com/kewisch/gdata-provider/commit/8a86dd5cbd833a70813872995f4b2ed64681759e)), closes [#191](https://github.com/kewisch/gdata-provider/issues/191)
* Use oob mechanism for OAuth instead ([6b243be](https://github.com/kewisch/gdata-provider/commit/6b243be734707ab07758ca2057ebd5ad56cbe1f6)), closes [#240](https://github.com/kewisch/gdata-provider/issues/240)

### Features

* Go back to the session page if auth dialog is canceled ([4969f6f](https://github.com/kewisch/gdata-provider/commit/4969f6f1f4e0bfde97c83ab3ee9d2c12d8a01ef1))
* Invalidate session when password is removed ([abd16b1](https://github.com/kewisch/gdata-provider/commit/abd16b100636fec8b99dfac555f5317ff21e827d))
* Options UI for invitations options ([abc3a41](https://github.com/kewisch/gdata-provider/commit/abc3a417dcf7b6940f599140523f216eca846af4))
* Roundtrip color id so it is not removed ([90121ab](https://github.com/kewisch/gdata-provider/commit/90121abfbf3d25fd1119f9d665de4f6ec939c8d2)), closes [#236](https://github.com/kewisch/gdata-provider/issues/236)

**Comment by kewisch** [comment]
I would appreciate if folks could give it a test. I'm going to let this sink in for a few days, then spin a release: [gdata-provider-91.0.2pre4.xpi.zip](https://github.com/kewisch/gdata-provider/files/7316513/gdata-provider-91.0.2pre4.xpi.zip)

**Reply by rachyandco** [reply]
Thanks. 

installation failed, because provider is not compatible with thunderbird 78.13.0

**Reply by rebel2k** [reply]
I just downloaded and installed. No issues so far on an up-to-date TB version. I'll keep you posted if anything comes up wrong with my tests.

**Reply by EkriirkE** [reply]
Got my calendars back, and dismissed all the backlog events but one is persisting and leaving the calendar in a "momentarily not available" state, and calendar emails say there are no writeable calendars configured for invitations.  I verified it's not setup as read-only

**Reply by mac-mac** [reply]
Here, it looks like the problem with deletion of tasks outside of TB still persists. My original post [#72](https://github.com/kewisch/gdata-provider/issues/72#issuecomment-764909993).
Thunderbird 91.2.0 (32 bit)
Google Calendar Provider 91.0.2

**Reply by kewisch** [reply]
@rachyandco Yes, you need to update to Thunderbird 91 for this version of the Provider.

@EkriirkE Do you have the three checkboxes set in the preferences? This may help. Also, make sure you are subscribed to the primary calendar, you cannot accept invitations to any secondary calendars.

@mac-mac Unfortunately there is no way to do push updates for desktop applications using the gdata API. The default sync interval is 30 minutes or so, which I can't lower too much as it would cause the Provider to hit rate limiting that affects everyone. Does it update after the configured interval?

**Reply by mac-mac** [reply]
No, it's not the problem with push updates. Tasks deleted from outside of TB, do not disappear in TB at all, even after manual synchronization.
There is another post by user Sebastian1969 (https://github.com/kewisch/gdata-provider/issues/72#issuecomment-660942040) where he describes problem this way:
> 
> Generating a new task in Thunderbird: OK, synchronized and visible in Google Android App and Online 
Deleting an existing task in Thunderbird: OK, synchronized and visible in Google Android App and Online
> 
> Generating a new task in Google Android App: OK, synchronized and visible in Thunderbird 
Deleting an existing task in Google Android App: **NOT** synchronized, item still visible in Thunderbird.

...and that's the exact same scenario I'm having.

**Reply by andrewgdotcom** [reply]
I'm running TB 91.2.0 on MacOS and Betterbird 91.1.0 on Linux. In both cases, the three new plugin settings do not persist; if I leave the plugin preferences and go back in they have been cleared. I can see no way to save them

**Reply by andrewgdotcom** [reply]
... and I've just suffered data loss again; all attendees to all my calendar events have been removed.

**Reply by EkriirkE** [reply]
My issue was related to not having a promary email selected for the calendar per other threads on this error.  Google calendars don't have that option...  I fixed my issues by disabling the plugin which allowed me to select my email address in the calendar properties, then re-enabled the plugin and invitations worked.
